tests/test_maker_utils.py

```diff
@@ -0,0 +1,78 @@
+import asdf
+import pytest
+
+from roman_datamodels import maker_utils, stnode
+from roman_datamodels.maker_utils import _ref_files as ref_files
+
+
+@pytest.mark.parametrize("node_class", stnode.NODE_CLASSES)
+def test_maker_utility_implemented(node_class):
+    """
+    Confirm that a subclass of TaggedObjectNode has a maker utility.
+    """
+    instance = maker_utils.mk_node(node_class)
+    assert isinstance(instance, node_class)
+
+
+@pytest.mark.parametrize("node_class", stnode.NODE_CLASSES)
+def test_instance_valid(node_class):
+    """
+    Confirm that a class's maker utility creates an object that
+    is valid against its schema.
+    """
+    with asdf.AsdfFile() as af:
+        af["node"] = maker_utils.mk_node(node_class)
+        af.validate()
+
+
+@pytest.mark.parametrize("node_class", [c for c in stnode.NODE_CLASSES if issubclass(c, stnode.TaggedObjectNode)])
+def test_no_extra_fields(node_class, manifest):
+    instance = maker_utils.mk_node(node_class)
+    instance_keys = set(instance.keys())
+
+    schema_uri = next(t["schema_uri"] for t in manifest["tags"] if t["tag_uri"] == instance.tag)
+    schema = asdf.schema.load_schema(schema_uri, resolve_references=True)
+
+    schema_keys = set()
+    subschemas = [schema]
+    if "allOf" in schema:
+        subschemas.extend(schema["allOf"])
+    for subschema in subschemas:
+        schema_keys.update(subschema.get("properties", {}).keys())
+
+    diff = instance_keys - schema_keys
+    assert len(diff) == 0, "Dummy instance has extra keys: " + ", ".join(diff)
```

Comment:
These are basically the same thing as the factory tests use, adapted for the maker utilities